python/ray/serve/controller.py

```diff
@@ -420,13 +436,46 @@ def deploy_app(
                 "previous request."
             )
 
-        deployment_override_options = config.dict(
-            by_alias=True, exclude_unset=True
-        ).get("deployments", [])
-
         self.config_deployment_request_ref = run_graph.options(
             runtime_env=config.runtime_env
-        ).remote(config.import_path, config.runtime_env, deployment_override_options)
+        ).remote(
+            config.import_path,
+            config.runtime_env,
+            deployment_override_options,
+            version_dict,
+        )
+
+    def get_version_of_new_deployments(
```

Comment:
Can we make this function take in two configs instead of two deployment dictionaries? It would help clarify that this feature is specifically for deploying with a config. It would also help reduce the dictionary processing code in `deploy_app`.